asab/library/providers/zookeeper.py

```diff
@@ -20,89 +22,115 @@ def __init__(self, app, path):
 		super().__init__(app, path)
 		self.App = app
 		self.Path = path
+
 		url_pieces = urllib.parse.urlparse(self.Path)
 		self.BasePath = url_pieces.path.lstrip("/")
-		# remove leading.
-		if self.BasePath.endswith("/"):
-			self.BasePath = self.BasePath.rstrip("/")
-		zksvc = self.App.get_service("asab.ZooKeeperService")
+
+		while self.BasePath.endswith("/"):
+			self.BasePath = self.BasePath[:-1]
+
 		# Initialize ZooKeeper client
-		self.ZookeeperContainer = asab.zookeeper.ZooKeeperContainer(
+		zksvc = self.App.get_service("asab.ZooKeeperService")
+		self.ZookeeperContainer = ZooKeeperContainer(
 			zksvc,
 			config_section_name='',
 			z_path=self.Path
 		)
+
 		self.Zookeeper = None
 		self.DisabledPaths = None
-		self.FileExtentions = frozenset({'.yaml'})
+
 		self.App.PubSub.subscribe("ZooKeeperContainer.started!", self._on_zk_ready)
 		self.Zookeeper = self.ZookeeperContainer.ZooKeeper
 
-	async def _on_zk_ready(self, event_name, zkcontainer):
-		if zkcontainer == self.ZookeeperContainer:
-			self.DisabledPaths = await self._load_disabled()
-
-	async def _load_disabled(self):
-		try:
-			disabled_data = await self.read(".disabled.yaml")
-			return yaml.safe_load(disabled_data)
-		except Exception as e:
-			L.error("The following exception occurred while loading disabled list: '{}'.".format(e))
-			return None
 
 	async def finalize(self, app):
-		# close client
+		"""
+		The `finalize` function is called when the application is shutting down
+		"""
 		await self.Zookeeper._stop()
 
+
+	async def _on_zk_ready(self, event_name, zkcontainer):
+		"""
+		When the Zookeeper container is ready, set the self.Zookeeper property to the Zookeeper object.
+		"""
+		if zkcontainer == self.ZookeeperContainer:
+			self.Zookeeper = self.ZookeeperContainer.ZooKeeper
+
+
 	async def read(self, path):
 		if self.Zookeeper is None:
 			L.warning("Zookeeper Client has not been established (yet). Cannot read {}".format(path))
```

Comment:
@ateska why are we doing this check ? They are in every method

`if self.Zookeeper is None:`

These checks are necessary?

Reply:
Yes.
In some cases, the Zookeeper connection is not ready yet, b/c the connection is asynchronous.
When this happens, Kazoo provides very misleading exception, so we fix that here.

It is mainly to be developer friendly, not a production issue.